src/Microsoft.Data.Analysis/TextFieldParser.cs

```diff
@@ -0,0 +1,642 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Data.Analysis
+{
+    public enum FieldType
+    {
+        Delimited,
+        FixedWidth
+    }
+
+    public class TextFieldParser : IDisposable
+    {
+        private delegate int ChangeBufferFunction();
+
+        private bool _disposed;
+
+        private TextReader _reader;
+
+        private string[] _commentTokens = new string[0];
+
+        private long _lineNumber = 1;
+
+        private bool _endOfData;
+
+        private string _errorLine = "";
+
+        private long _errorLineNumber = -1;
+
+        private FieldType _textFieldType = FieldType.Delimited;
+
+        private int[] _fieldWidths;
+
+        private int[] _fieldWidthsCopy;
+
+        private string[] _delimiters;
+
+        private string[] _delimitersCopy;
+
+        private int[] _whitespaceCodes = new int[] { '\u0020' };
+
+        private Regex _whiteSpaceRegEx = new Regex("\\s", RegexOptions.CultureInvariant);
+
+        private bool _trimWhiteSpace = true;
+
+        private int _position;
+
+        private int _peekPosition;
+
+        private int _charsRead;
+
+        private const int DEFAULT_BUFFER_LENGTH = 4096;
+
+        private char[] _buffer = new char[DEFAULT_BUFFER_LENGTH];
+
+        private bool _hasFieldsEnclosedInQuotes = true;
+
+        private int _maxBufferSize = 10000000;
+
+        private bool _leaveOpen;
+
+        private char[] newLineChars = Environment.NewLine.ToCharArray();
+
+        public string[] CommentTokens
+        {
+            get
+            {
+                return _commentTokens;
+            }
```

Comment:
```suggestion
            get => _commentTokens
```
? Makes the code shorter.
On other places as well.